### PR TITLE
Fix for SI-6537, inaccurate unchecked warning.

### PR DESCRIPTION
**Pull request by paulp**
I found a more direct expression of the unchecked logic,
which should be much easier for others to verify.  But the
bug being fixed here is that the unchecked checking happens
too early, and the sealed children of a symbol are not yet
visible if it is being simultaneously compiled.

### CODE REVIEW

**Comment by paulp** [comment]
Review by @adriaanm

File: src/compiler/scala/tools/nsc/typechecker/Checkable.scala [hunk 1]
```diff
@@ -165,41 +165,43 @@ trait Checkable {
...
     )
-
-    /** Given classes A and B, can it be shown that nothing which is
-     *  an A will ever be a subclass of something which is a B? This
-     *  entails not only showing that !(A isSubClass B) but that the
-     *  same is true of all their subclasses.  Restated for symmetry:
-     *  the same value cannot be a member of both A and B.
+    /** Is it impossible for the given symbols to be parents in the same class?
+     *  This means given A and B, can there be an instance of A with B? This is the
+     *  case if neither A nor B is a subclass of the other, and one of the following
```
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/typechecker/Checkable.scala:185]: i think there's a negation missing somewhere

Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/Checkable.scala:185]: "can there be an instance of A with B?" asks the opposite question as the method, but I thought it was apparent from context.  (Isn't there always a negation missing somewhere? Daring comment!)

Comment (paulp) [inline src/compiler/scala/tools/nsc/typechecker/Checkable.scala:185]: Get it, "apparent"

Comment (adriaanm) [inline src/compiler/scala/tools/nsc/typechecker/Checkable.scala:185]: I don't, apparently, but I'll let it slide. Unqualified LGTM then.

Comment (adriaanm) [inline src/compiler/scala/tools/nsc/typechecker/Checkable.scala:185]: "This is the case," you say? But what is the case?

**Comment by adriaanm** [comment]
LGTM

**Pull request merged**